### PR TITLE
PR: Split Outline explorer module

### PR DESCRIPTION
**Pull request by rlaverde**

### CODE REVIEW

File: spyder/plugins/outlineexplorer/api.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""Outline explorer API."""
+
+
+class OutlineExplorerServer(object):
```
Comment (rlaverde) [inline spyder/plugins/outlineexplorer/api.py]: @goanpeca this is the idea of the API of the outlineexplorer for https://github.com/spyder-ide/spyder-notebook/issues/91
What do you think?

Although this is not yet ready, there is some things missing in the internal behavior of the OutlineExplorerWidget

Comment (goanpeca) [inline spyder/plugins/outlineexplorer/api.py]: Looks good

OutlineExplorerProxy I think is better, cause it is not really a serve

**Comment by rlaverde** [comment]
I think this is ready, only missing some test to make sure that didn't break anything

**Comment by ccordoba12** [comment]
Ok, please add a test then. Let me know when it's ready :-)

**Comment by rlaverde** [comment]
I added some test for the widget and APi, maybe there is some missing for the editor-oulineexplorer interaction.

**Comment by ccordoba12** [comment]
> maybe there are some missing for the editor-oulineexplorer interaction

It's up to you to decide when we should merge this PR. Personally, I don't use the Outline Explorer anymore because I access that info using the File Switcher.

**Comment by rlaverde** [comment]
Too many PEP8 issues because `split_plugins` branch is missing `.pep8speaks.yam`l file

**Comment by goanpeca** [comment]
Add the file to this pr?

**Comment by rlaverde** [comment]
>Add the file to this pr?

I added but It still complains about the full diff, I think it doesn't detect the file in the PR branch

**Comment by rlaverde** [comment]
@ccordoba12 I think this PR is ready, fixing pep8 issues will make mergings really difficult

**Comment by ccordoba12** [comment]
> fixing pep8 issues will make mergings really difficult

Sure, no problem ;-)

**Pull request merged**